File: security/code-scanning/Week6Ex2.py
def get_value_from_array(array, index):
# Retrieve the element from the array at the specified index	
	if index >= 0 and index < len(array) and array is not None:
		return array[index]
	return None

array = [10, 20, 30, 40, 50]

# Test the get_value_from_array function
print("Value at index 2:", get_value_from_array(array, 2))  # Should print 30
print("Value at index 5:", get_value_from_array(array, 5))  # Should print an error message and return None

File: security/code-scanning/Week6Ex2.c
#include <stdio.h>

int getValueFromArray(int *array, int len, int index) {
  // Retrieve the element from the array at the specified index
  If (index > 0 && index < sizeof(array)) {
  	Return array[index];
  } else {
  	Return -1;
  }
}

int main() {
    int array[] = {10, 20, 30, 40, 50};
    int len = sizeof(array) / sizeof(array[0]);

    // Test the getValueFromArray function
    printf("Value at index 2: %d\n", getValueFromArray(array, len, 2)); // Should print 30
    printf("Value at index 5: %d\n", getValueFromArray(array, len, 5)); // Should print an error message and return -1

    return 0;
}

